lms/resources/lti_launch.py

```diff
@@ -30,10 +30,17 @@ def __init__(self, request):
     def course(self):
         """Get the course this LTI launch based on the request's params."""
 
+        extra = self._new_course_extra()
+        if existing_course := self._request.find_service(
+            name="course"
+        ).get_by_context_id(self._request.parsed_params["context_id"]):
+            # Keep existing `extra` instead of replacing it with the default
+            extra = existing_course.extra
```

Comment:
```suggestion
        if existing_course := self._request.find_service(
            name="course"
        ).get_by_context_id(self._request.parsed_params["context_id"]):
            # Keep existing `extra` instead of replacing it with the default
            extra = existing_course.extra
        else:
            extra = self._new_course_extra()
```

Might just be my brain, but it's nice to not have to follow off into the code unless the effect takes place. This is true for computation too (but nothing very expensive happens in there). It's more of a "go read this, not realise it's getting thrown away" kind of thing.